src/indra_cogex/client/queries.py

```diff
@@ -1140,6 +1144,246 @@ def get_stmts_for_stmt_hashes(
+    """
+
+    if agent_name and agent_role:
+        if agent_role.lower() == "subject":
+            match_clause = f"(a:BioEntity {{name: $agent_name}})-[r:indra_rel"
+            match_direction = "]->(b:BioEntity)"
+        elif agent_role.lower() == "object":
+            match_clause = f"(a:BioEntity)-[r:indra_rel"
+            match_direction = f"]->(b:BioEntity {{name: $agent_name}})"
+        else:
+            raise ValueError("agent_role must be 'subject' or 'object'")
+    else:
+        match_clause = "(a:BioEntity)-[r:indra_rel"
+        match_direction = "]->(b:BioEntity)"
+
+    rel_constraints = []
+    if rel_type:
+        rel_constraints.append("stmt_type: $rel_type")
+
+    if rel_constraints:
+        match_clause += " {" + ", ".join(rel_constraints) + "}"
+
+    query = f"MATCH p = {match_clause}{match_direction}"
+
+    where_clauses = []
+    if stmt_source:
+        where_clauses.append(f'r.source_counts CONTAINS \'"{stmt_source}":\'')
+    if where_clauses:
+        query += " WHERE " + " AND ".join(where_clauses)
+
+    query += " RETURN p LIMIT $limit"
+
+    params = {
+        "agent_name": agent_name,
+        "rel_type": rel_type,
+        "limit": limit
+    }
+
+    logger.info(f"Running query with constraints: rel_type={rel_type}, "
+                f"source={stmt_source}, agent_name={agent_name}, "
+                f"agent_role={agent_role}, limit={limit}")
+    rels = client.query_relations(query, **params)
+    stmts = indra_stmts_from_relations(rels, deduplicate=True)
+
+    if evidence_limit:
+        stmts = enrich_statements(
+            stmts,
+            client=client,
+            evidence_limit=evidence_limit,
+        )
+
+    return stmts
+
+@autoclient()
+def get_stmts_for_agent_type(
+    agent_name: str,
+    agent_role: str,
+    limit: Optional[int] = 10,
+    *,
+    client: Neo4jClient,
+    evidence_limit: Optional[int] = None
+) -> List[Statement]:
+    """Return the statements for a given agent based on its role
+
+    Parameters
+    ----------
+    agent_name : str
+        The name of the agent (e.g.,"MEK").
+    agent_role : str
+        The role of the agent in the interaction, either "subject" or "object".
+    limit : Optional[int], default: 10
+
+    client : Neo4jClient
+
+    evidence_limit : Optional[int], default: None
+
+    Returns
+    -------
+    List[Statement]
+        A list of statements where the given agent acts as the subject or object.
+    """
+
+    # Construct the query based on the role of the agent
+    if agent_role.lower() == "subject":
+        query = """
+        MATCH p = (a:BioEntity {name: $agent_name})-[r:indra_rel]->(b:BioEntity)
+        RETURN p LIMIT $limit
+        """
+    elif agent_role.lower() == "object":
+        query = """
+        MATCH p = (a:BioEntity)-[r:indra_rel]->(b:BioEntity {name: $agent_name})
+        RETURN p LIMIT $limit
+        """
+    else:
+        raise ValueError("agent_role must be 'subject' or 'object'")
```

Comment:
I think it should be possible to allow the agent to be either of `subject` or `object` as well. That query would either be a non-directed query: `()-[]-()` or a union of two directed queries `()-[]->()` and `()<-[]-()` and then ensuring there are no duplicates in the results.